File: code/ch13/13.2.best_time_to_buy_and_sell_stock_II.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) <= 1:
            return 0
        max_profit = 0
        for i in range(1, len(prices)):
            if prices[i] - prices[i - 1] > 0:
                max_profit += prices[i] - prices[i - 1]
        return max_profit

File: code/ch13/13.2.best_time_to_buy_and_sell_stock_II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit = 0;
        int size = prices.size();

        for (int i = 1; i < size; i++) {
            if (prices[i] > prices[i - 1])
                maxprofit += prices[i] - prices[i - 1];
        }
        
        return maxprofit;
    }
};
